test/src/test/java/org/apache/accumulo/test/functional/TableChangeStateIT.java

```diff
@@ -0,0 +1,549 @@
+
+    return tstate;
+  }
+
+  /**
+   * Executes TableOperations online command in a separate thread and blocks the current thread until command is complete. The status and time to complete is
+   * returned in {@code OnlineStatus}
+   *
+   * @param tableName
+   *          the table to set online.
+   * @return status that can be used to determine if running and timing information when complete.
+   */
+
+  private OnlineStatus onLine(String tableName) {
+
+    OnLineThread onLineCmd = new OnLineThread(tableName);
+    Thread onLineThread = new Thread(onLineCmd);
+    onLineThread.start();
+
+    int count = 0;
+
+    while (onLineCmd.getStatus().isRunning()) {
+
+      if ((count++ % 10) == 0) {
+        log.trace("online operation blocked, waiting for it to complete.");
+      }
+
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException ex) {
+        // reassert interrupt
+        Thread.currentThread().interrupt();
+      }
+    }
+
+    return onLineCmd.getStatus();
+  }
+
+  /**
+   * Create the provided table and populate with some data using a batch writer. The table is scanned to ensure it was populated as expected.
+   *
+   * @param tableName
+   *          the name of the table
+   */
+  private void createData(final String tableName) {
+
+    try {
+
+      Connector connector = getConnector();
+
+      // create table.
+      connector.tableOperations().create(tableName);
+      BatchWriter bw = connector.createBatchWriter(tableName, new BatchWriterConfig());
+
+      // populate
+      for (int i = 0; i < NUM_ROWS; i++) {
+        Mutation m = new Mutation(new Text(String.format("%05d", i)));
+        m.put(new Text("col" + Integer.toString((i % 3) + 1)), new Text("qual"), new Value("junk".getBytes(UTF_8)));
+        bw.addMutation(m);
+      }
+      bw.close();
+
+      long startTimestamp = System.nanoTime();
+
+      Scanner scanner = connector.createScanner(tableName, Authorizations.EMPTY);
+      int count = 0;
+      for (Map.Entry<Key,Value> elt : scanner) {
+        String expected = String.format("%05d", count);
+        assert (elt.getKey().getRow().toString().equals(expected));
+        count++;
+      }
+
+      log.trace("Scan time for {} rows {} ms", NUM_ROWS, TimeUnit.MILLISECONDS.convert((System.nanoTime() - startTimestamp), TimeUnit.NANOSECONDS));
+
+      scanner.close();
+
+      if (count != NUM_ROWS) {
+        throw new IllegalStateException(String.format("Number of rows %1$d does not match expected %2$d", count, NUM_ROWS));
+      }
+    } catch (AccumuloException | AccumuloSecurityException | TableNotFoundException | TableExistsException ex) {
+      throw new IllegalStateException("Create data failed with exception", ex);
+    }
+  }
+
+  /**
+   * Provides timing information and online command status results for online command executed asynchronously. The
+   */
+  private static class OnlineStatus {
```

Comment:
Since multiple instance vars are being written and read by multiple threads, should probably synchronize all methods in this class.